core/src/vote_redundancy_checker.rs

```diff
@@ -0,0 +1,146 @@
+use log::*;
+use solana_runtime::bank::Bank;
+use solana_sdk::{clock::Slot, pubkey::Pubkey, transaction::Transaction};
+use solana_vote_program::vote_transaction;
+use std::{collections::HashMap, sync::Arc};
+
+#[derive(Default, Debug)]
+pub struct VoteRedundancyChecker {
+    latest_voted_slots: HashMap<Pubkey, Slot>,
+}
+
+impl VoteRedundancyChecker {
+    pub fn new() -> Self {
+        Self {
+            latest_voted_slots: HashMap::new(),
+        }
+    }
+
+    pub fn check(
+        &mut self,
+        transaction: &Transaction,
+        bank: &Arc<Bank>,
+    ) -> Result<(), &'static str> {
+        if let Some((vote_account_pubkey, vote, _hash)) =
+            vote_transaction::parse_vote_transaction(transaction)
```

Comment:
This should probably be done outside of holding the lock around the `VoteRedundancyChecker` and the results of the parsing passed in here